pkg/controller/storagecluster/update.go

```diff
@@ -661,7 +661,17 @@ func (c *Controller) getPodAvailability(
 			if len(storageNode.SchedulerNodeName) == 0 {
 				nonK8sStorageNodes = append(nonK8sStorageNodes, storageNode)
 			} else {
-				storageNodeMap[storageNode.SchedulerNodeName] = storageNode
+				if existing, ok := storageNodeMap[storageNode.SchedulerNodeName]; ok {
+					// During a k8s upgrade, it may happen that a new node is started on the same k8s host using
```

Comment:
Shouldn't we verify that it's storageless and not do this for storage node?

Reply:
Thanks for pointing this out @nrevanna .

After the offline discussion, I have added a comment and added more details for the scenario in https://purestorage.atlassian.net/browse/PWX-38505. We will handle this later if we find out this scenario is possible and we can reproduce it.